core/Mutations/DebGoyalMutation.py

```diff
@@ -0,0 +1,56 @@
+import numpy as np
+from core import Mutation, Individual, Population
+
+
+class DebGoyalMutation(Mutation):
+    def __init__(self, n: int, delta_max: float, probability: float = 0):
+        """
+        Constructor for the Deb & Goyal Mutation (DGM) class.
+
+        :param probability: Probability of mutation (float between 0 and 1).
+        :param n: Distribution index for mutation.
+        :param delta_max: Maximum allowed perturbation.
+        :raises ValueError: If the distribution index is less than 1.
+        """
+        super().__init__(probability)
+        if int(n) < 1:
+            raise ValueError("The distribution index must be at least 1 to perform the DebGoyalMutation operation.")
+        self.n = n
+        self.delta_max = delta_max
+
+    def mutate(self, population_parent: Population) -> Population:
+        """
+        Perform the Deb & Goyal Mutation operations for the entire population.
+
+        :param population_parent: The population to perform the mutation operation on.
+        :returns: The mutated population.
+        """
+        for individual in population_parent.population:
+            self._mutate(individual, population_parent)
+
+        return population_parent
+
```

Comment:
Are you sure you need to overload that function? The one from abstract class isn't enough? I ask that question since currently I don't have access to the book with operators xD

Reply:
Yes, I think it needs to be done. According to the pseudocode in the book, if I didn't override that public method, it would perform the randomization twice, which would lower the chance of mutation. This mutation behaves a bit differently from others, so I perform the randomization inside the private method. Do you have access to the book, though? Mateusz or I could send you screenshots if needed.